crates/sargon/src/system/sargon_os/sargon_os_transactions.rs

```diff
@@ -0,0 +1,382 @@
+use crate::prelude::*;
+use std::time::Duration;
+
+// ==================
+// Submit Transaction
+// ==================
+#[uniffi::export]
+impl SargonOS {
+    /// Submits a notarized transaction payload to the network.
+    pub async fn submit_transaction(
+        &self,
+        notarized_transaction: NotarizedTransaction,
+    ) -> Result<IntentHash> {
+        let network_id = self.current_network_id()?;
+        let gateway_client = GatewayClient::new(
+            self.clients.http_client.driver.clone(),
+            network_id,
+        );
+        gateway_client
+            .submit_notarized_transaction(notarized_transaction)
+            .await
+    }
+
+    /// Submits a compiled transaction payload to the network.
+    pub async fn submit_compiled_transaction(
+        &self,
+        compiled_notarized_intent: CompiledNotarizedIntent,
+    ) -> Result<IntentHash> {
+        self.submit_transaction(compiled_notarized_intent.decompile())
+            .await
+    }
+}
+
+// ==================
+// Poll Transaction Status
+// ==================
+#[uniffi::export]
+impl SargonOS {
+    /// Polls the state of a Transaction until we can determine its `TransactionStatus`.
+    pub async fn poll_transaction_status(
+        &self,
+        intent_hash: IntentHash,
+    ) -> Result<TransactionStatus> {
+        let network_id = self.current_network_id()?;
+        let gateway_client = GatewayClient::new(
+            self.clients.http_client.driver.clone(),
+            network_id,
+        );
+
+        // The delay increment is set to 1 second in production, but 0 seconds in tests.
+        // This will make the tests run with 0s delay, while the production code will have a 2s delay on first call,
+        // a 3s delay on second call, 4s on third and so on.
+        #[cfg(test)]
+        const DELAY_INCREMENT: u64 = 0;
+        #[cfg(not(test))]
+        const DELAY_INCREMENT: u64 = 1;
+
+        let mut delay_duration = DELAY_INCREMENT;
+        loop {
```

Comment:
we might wanna include a timeout value to this poll function? where we typically would use `30sec` or so by hosts ?

Reply:
what would be the benefit of having a timeout? We don't have one on current implementation ([iOS](https://github.com/radixdlt/babylon-wallet-ios/blob/main/RadixWallet/Clients/SubmitTransactionClient/SubmitTransactionClient%2BLive.swift#L23-L51)), and I don't think there is a reason to have one. 
If the network is congested and takes longer than 30s to process, our requests to the gateway would be every time more separated in time each from each other (meaning we won't saturate our GW).

Reply:
Yeah maybe not needed. Just feels right :) 

I think gateway team would prefer it too. A max number of polls